climada/hazard/tc_tracks_synth.py

```diff
@@ -274,87 +647,255 @@ def _one_rnd_walk(track, nb_synth_tracks, max_shift_ini, max_dspeed_rel, max_ddi
     cutoff_track_ids_tc : List of str
         List containing information about the tracks that were cut off at high
         latitudes with wind speed of TC category 2-5.
-    curoff_track_ids_ts : List of str
+    cutoff_track_ids_ts : List of str
         List containing information about the tracks that were cut off at high
         latitudes with a wind speed up to TC category 1.
     """
+
+    # on_land parameters of historical track will be required for each synthetic track
+    if land_geom is not None:
+        climada.hazard.tc_tracks.track_land_params(track, land_geom, land_params='on_land')
+
     ens_track = list()
-    n_dat = track.time.size
-    n_seg = n_dat - 1
-    xy_ini = max_shift_ini * (2 * rnd_vec[:2 * nb_synth_tracks].reshape((2, nb_synth_tracks)) - 1)
-    [dt] = np.unique(track['time_step'])
-
-    ens_track.append(track)
-    cutoff_track_ids_ts = []
-    cutoff_track_ids_tc = []
+    ens_track.append(track.copy(True))
+
+    # calculate historical track values that are used for synthetic track modelling
+    if land_geom is not None:
+        # compute minimum pressure occurring on or after each timestep
+        if int(os.getenv('TRACKGEN_TEST_TARGET_PRESSURE_OVER_SEA')):
```

Comment:
Can you explain the rationale behind this @ChrisFairless ?
My understanding is: you only consider the minimum central pressure of all ocean points only after each timestep (including also the last central pressure value, which I guess is to make sure one value is available for each time step).
Why is that? In which case would you expect minimum future central pressure to be lower (i.e. stronger) if we include also land points? And if this case arises, why not accounting for it?